### PR TITLE
Glossary file.

### PR DESCRIPTION
**Pull request by yhontyk**
### What does this PR do?
Adds content for Glossary concept module.

To-do:

- [x] determine placement
- [ ] verify content for downstream 

### Specify the version of the product this PR applies to.

### CODE REVIEW

File: src/main/pages/che-7/overview/con_glossary.adoc [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+= {prod-short} Glossary
+
+:context: {prod-id-short}-glossary
+
+== Naming
+{prod}:: {prod} is a developer platform for the cloud that provides an in-browser IDE. {prod-short} tools can be extended by adding development services, such as language servers, debug adapters, or editors (IDEs) packaged as containers. 
+
+Che-Theia:: Che-Theia is the default editor component of a {prod-short} workspace. It is a {prod-short}-specific customization of the Eclipse Theia editor. Che-Theia can be extended using Che-Theia plugins, which are packaged as containers for ease of deployment and use. Che-Theia plugins are mostly compatible with Visual Studio extensions.
+
+{prod-short} Workspace:: {prod-short} Workspace is a container-based development environment managed by {prod}. Every {prod-short} workspace is defined by a devfile. A {prod-short} workspace is composed by an editor, some runtime and build containers and other development tools running inside containers.
```
Comment (kowen-rh) [inline src/main/pages/che-7/overview/con_glossary.adoc]: "Composed by" or "composed of"?

File: src/main/pages/che-7/overview/con_glossary.adoc [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+== Terminology
+Organization:: Organizations in {prod} allow administrators to group {prod} users and allocate resources. The system administrator controls and allocates resources and permissions within the administrator dashboard.
+
+Editor:: Editor is a web application that is used as an editor in a workspace.
```
Comment (kowen-rh) [inline src/main/pages/che-7/overview/con_glossary.adoc]: "Editor is" is unnecessary. In a definition list, it's implied. :)

File: src/main/pages/che-7/overview/con_glossary.adoc [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+Plugin:: {prod-short} plugins are services that extend {prod-short} workspace capabilities. {prod-short} plugins are packaged as containers. Plugins are extensions of an editor or a service running in the container. For example, the Che-Theia editor is compatible with Visual Studio Code extensions. 
+//TODO See for a diagram of Che extensibility architecture. 
+Both {prod-short} plugins and editors are distributed through the {prod-short} plugin registry. 
+ 
+Workspace:: A container based development environment managed by Eclipse Che. Every  {prod-short} workspace is defined by a Devfile. A  {prod-short} workspace can be composed by an editor, some plugins and runtime containers. Workspace runtime containers can be defined as simple container images or as Kubernetes and OpenShift resources. A {prod-short} Workspace can be associated with source code projects hosted on a remote CVS server. A {prod-short} Workspace can contain the definition of one or more commands such as `run`, `build`, or `debug`.
```
Comment (kowen-rh) [inline src/main/pages/che-7/overview/con_glossary.adoc]: There's a double-space in "Every {prod-short} workspace is defined...."

"Composed by" or "composed of"?

Comment (kowen-rh) [reply src/main/pages/che-7/overview/con_glossary.adoc]: Also, "Devfile" is capitalized here, but not capitalized elsewhere within this PR. Should it be capitalized or not?

File: src/main/pages/che-7/overview/con_glossary.adoc [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+Devfile:: A devfile is a workspace configuration template. Devfiles are used to create workspaces in the Dashboard. A devfile includes meta information such as scope, tags, components, description, name, and identification. VS Code API is one of the plugin APIs that can be used in Theia.
```
Comment (kowen-rh) [inline src/main/pages/che-7/overview/con_glossary.adoc]: Should "meta information" be "metadata"?

Also, no need for "A devfile is", similar to above. It's implied by the definition list. :)

File: src/main/pages/che-7/overview/con_glossary.adoc [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+Factory:: A factory is a template that contains the configuration to automate the generation of a new workspace using a factory identifier or a devfile added to the URL of the IDE. Use factories to create replicas of existing workspaces or to automate the provisioning of statically or dynamically defined workspaces.
```
Comment (kowen-rh) [inline src/main/pages/che-7/overview/con_glossary.adoc]: No need for "A factory is", similar to above. :)

**Comment by kowen-rh** [comment]
Also, I just noticed that the commit message is simply "Initial commit." Could you provide a more descriptive commit message? :)

**Review by MichalMaler — APPROVED** [review]

**Pull request merged**